src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
```

Comment:
Indeed; the very issue that held up gcc adopting it for a long time. With DDA's simple file structure, it doesn't seem like it could ever cause trouble, but I'll stick to include guards for DDA I guess.